### PR TITLE
Update tsc and fix linux install

### LINKED ISSUE #22: Install on Ubuntu
I am trying to install kudusync via ('sudo npm install kudusync -g') on Ubuntu 14.04 and get this error message.
`npm ERR! Error: ENOENT, chmod '/usr/local/lib/node_modules/kudusync/bin/kudusync'`.
node version is 0.10.25
npm version is 1.3.10

Please let me know if you need more information

### PR DESCRIPTION
**Pull request by albertjan**
fixes #22 

linuxinstall

### CODE REVIEW

File: bin/kudusync.js [hunk 1]
```diff
@@ -1,35 +1,39 @@
+///<reference path='..\typings\node.d.ts'/>
+///<reference path='..\typings\q.d.ts'/>
 var fs = require('fs');
 var pathUtil = require('path');
 var Q = require('q');
 var minimatch = require('minimatch');
 var log = console.log;
-if(!fs.existsSync) {
```
Comment (amitapl) [inline bin/kudusync.js]: why remove the `{`?

File: bin/kudusync.js [hunk 2]
```diff
@@ -1,35 +1,39 @@
...
+// Workaround to support both APIs whether in node 0.6.* or 0.8.0 and higher.
+if (!fs.existsSync)
     fs.existsSync = pathUtil.existsSync;
-}
+///<reference path='header.ts'/>
```
Comment (amitapl) [inline bin/kudusync.js]: Can you move all references to the top?

File: lib/Manifest.ts [hunk 3]
```diff
@@ -5,7 +5,7 @@ class Manifest {
     private _files: string[];
 
     constructor () {
-        this._files = new string[];
```
Comment (amitapl) [inline lib/Manifest.ts]: why is the change to Array required?

File: lib/Utils.ts [hunk 4]
```diff
@@ -1,8 +1,8 @@
-///<reference path='ensure.ts'/>
+///<reference path='Ensure.ts'/>
```
Comment (amitapl) [inline lib/Utils.ts]: why is the casing change needed?

**Comment by amitapl** [comment]
Except for the `kuduSync` to `kudusync` change is there any other required change here?

**Comment by amitapl** [comment]
I will test and make sure everything works

File: lib/Utils.ts (hunk 4 above)
Comment (albertjan) [inline lib/Utils.ts]: It isn't. The file is called Ensure.ts so thats why I changed it.

File: lib/Manifest.ts (hunk 3 above)
Comment (albertjan) [inline lib/Manifest.ts]: The typescript compiler doesn't like `string[]`. It says:

> lib/manifest.ts(8,33): error TS1150: 'new T[]' cannot be used to create an array. Use 'new Array<T>()' instead.

**Comment by albertjan** [comment]
No none of the other changes are required for the linux install, I thought it'd be a good idea to update to the latest TypeScript compiler, and update the other packages.

**Comment by amitapl** [comment]
I'm reluctant on making the typescript update as I don't see any advantage in doing so at least until there is a change to make that requires it.

Do you know of any improvements (we need) it has on the result js script?

**Comment by albertjan** [comment]
I understand. There are no real reasons other then staying up to date. Functionaly the output __should__ be the same, and all tests pass. But I understand you don't want to risk loads of ppl running into something weird, just to update the compiler.

Do you feel the same about the dependencies?

**Comment by amitapl** [comment]
The dependencies seem safe

**Comment by albertjan** [comment]
Allright. I'll close this one and open a new PR. :smile:

**Pull request closed without merging**